packages/cli/internal/pkg/cli/account_activate.go

```diff
@@ -182,6 +201,28 @@ func (o accountActivateOpts) deployCoreInfrastructure(cdkAppPath string, environ
 	return displayProgress(progressStream, "Activating account...")
 }
 
+func (o *accountActivateOpts) validate() error {
+	if o.publicSubnets && o.vpcId != "" {
+		return o.generateValidationError(
+			fmt.Errorf("both %[1]q and %[2]q cannot be specified together, as %[2]q involves creating a minimal VPC", accountVpcFlag, publicSubnetsFlag),
+			"Remove one or both of these flags")
+	}
+
+	if o.subnets != nil && o.vpcId == "" {
```

Comment:
The reliable way to check that a slice is empty is to do `len(someSlice) == 0`. In this case it can be empty but not `nil`.